src/hash.rs

```diff
@@ -1,5 +1,10 @@
+//! Cryptographic hash functions
+//!
+//! The [`sha256`] function requires feature `sha2` or `ring`.
```

Comment:
Would it make sense to explain what happens if you load both crates?

Reply:
I think so. These two features conflict, i.e. compilation fails if they are both enabled. The features and their conflicts should be described in the table being added in `src/lib.rs`. But it could be good to mention this here somehow (e.g. "but not both").